src/nbl/video/COpenGL_Swapchain.h

```diff
@@ -207,12 +216,29 @@ class COpenGL_Swapchain final : public ISwapchain
 
             m_ctxCreatedCvar.notify_one();
 
+            switch (m_presentMode)
+            {
+            case ISurface::EPM_IMMEDIATE:
+                egl->call.peglSwapInterval(egl->display, 0);
+                break;
+            case ISurface::EPM_FIFO:
+                egl->call.peglSwapInterval(egl->display, 1);
+                break;
+            case ISurface::EPM_FIFO_RELAXED:
+                egl->call.peglSwapInterval(egl->display, -1);
+                break;
+            }
+
```

Comment:
what about MAILBOX present mode?

Reply:
I was told to skip it
here https://github.com/Devsh-Graphics-Programming/Nabla/pull/160#discussion_r676728146

Reply:
then you need to error out somehow if someone says "mailbox", cause right now its an unhandled case

@Crisspl MAILBOX has different semantics (tearing cannot be observed) to IMMEDIATE, its like a no-tear-no-Vsync. AFAIK even double buffering on OpenGL tears without vsync.

Reply:
well the default swap interval is already set
> The default swap interval is 1.

we can perhaps log a warning?

Reply:
no, just error out and refuse to create a swapchain with an unsupported presentation mode

Reply:
resolved